### PR TITLE
Use Vite instead of `create-react-app`

### PR DESCRIPTION
**Pull request by rygine**
This PR replaces `create-react-app` with Vite for a better developer experience.

With this major change comes other necessary changes:

* Renamed all `.js` => `.jsx`
* Removed `create-react-app` dependencies and utilities

Given the simplicity of this example, `react-router-dom` has also been removed.

Other notable changes:

* Updated `.gitignore`
* Upgraded all dependencies to latest version except `ethers`
* Removed unused React logo
* Removed testing utilities and dependencies
* Added support for TypeScript

# Next steps

In no particular order, the following changes will be made in the near future.

## Convert everything to TypeScript

This will update file extensions and add types where appropriate.

## Add prettier config and checks

To ensure consistent formatting, a prettier config will be added.

## Replace web3modal with rainbowkit

This change was recently made in the [xmtp-inbox-web](https://github.com/xmtp-labs/xmtp-inbox-web) repository for a better wallet connection experience.

## Use Tailwind for styling

This will add Tailwind and update current styling to integrate with it.

### CODE REVIEW

**Comment by neekolas** [comment]
Want to call out the "Convert everything to TypeScript" change. This was specifically written in Javascript based on developer feedback. This is a repo designed for more junior devs, many of which aren't familiar with Typescript. We all would have preferred to do this in TS, but decided against it to appeal to the lowest common denominator of developer.

**Comment by rygine** [comment]
> Want to call out the "Convert everything to TypeScript" change. This was specifically written in Javascript based on developer feedback. This is a repo designed for more junior devs, many of which aren't familiar with Typescript. We all would have preferred to do this in TS, but decided against it to appeal to the lowest common denominator of developer.

interesting. i saw [this issue](https://github.com/xmtp/xmtp-quickstart-react/issues/9) and figured it was something we wanted to add.

**Comment by neekolas** [comment]
> interesting. i saw [this issue](https://github.com/xmtp/xmtp-quickstart-react/issues/9) and figured it was something we wanted to add.

I hadn't seen that issue. I guess some developers want TS and some developers don't. We should have a conversation about this with @yash-luna and Eleanor and figure out who we care about more with this particular repo.

**Comment by rygine** [comment]
> I hadn't seen that issue. I guess some developers want TS and some developers don't. We should have a conversation about this with @yash-luna and Eleanor and figure out who we care about more with this particular repo.

ok, sounds good. we have the option with this setup if we decide to pursue it.

File: src/App.css [hunk 1]
```diff
@@ -2,37 +2,3 @@
   text-align: center;
 }
 
-.App-logo {
```
Comment (neekolas) [inline src/App.css:5]: Was all this CSS just unused?

**Comment by neekolas** [comment]
Tested this out locally and it all works as expected

**Review by neekolas — APPROVED** [review]

**Pull request merged**